synedrion/benches/zk_proofs.rs

```diff
@@ -0,0 +1,179 @@
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("verify", paillier_blum_modulus_proof_verify(rng));
+    }
+
+    fn bench_mul_star(c: &mut Criterion) {
+        use mul_star_proof::*;
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(EnvFilter::from_default_env())
+            .try_init();
+        let mut group = c.benchmark_group("MulStar proof");
+        group.sample_size(10);
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("prove", mul_star_proof_prove(rng));
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("verify", mul_star_proof_verify(rng));
+    }
+
+    fn bench_paillier_mul(c: &mut Criterion) {
+        use paillier_mul_proof::*;
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(EnvFilter::from_default_env())
+            .try_init();
+        let mut group = c.benchmark_group("Paillier multiplication proof");
+        group.sample_size(10);
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("prove", paillier_mul_proof_prove(rng));
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("verify", paillier_mul_proof_verify(rng));
+    }
+
+    fn bench_prm(c: &mut Criterion) {
+        use prm_proof::*;
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(EnvFilter::from_default_env())
+            .try_init();
+        let mut group = c.benchmark_group("Pedersen Ring params (prm) proof");
+        group.sample_size(10);
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("prove", prm_proof_prove(rng));
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("verify", prm_proof_verify(rng));
+    }
+
+    fn bench_sch(c: &mut Criterion) {
+        use sch_proof::*;
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(EnvFilter::from_default_env())
+            .try_init();
+        let mut group = c.benchmark_group("Schnorr (sch) proof");
+        group.sample_size(10);
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("prove", sch_proof_prove(rng));
+
+        let rng = rand_chacha::ChaCha8Rng::seed_from_u64(1234567890);
+        group.bench_function("verify", sch_proof_verify(rng));
+    }
+
+    criterion_group!(
+        benches,
+        bench_fac,
+        bench_aff_g,
+        bench_dec,
+        bench_enc,
+        bench_log_star,
+        bench_paillier_blum_modulus,
+        bench_mul_star,
+        bench_paillier_mul,
+        bench_prm,
+        bench_sch
+    );
+}
+
+// Running benchmarks without the test harness requires a main function at the top level, leading to
```

Comment:
I think you can avoid doing that by specifying `required-features = ["private-benches"]` for this benchmark in `Cargo.toml`. Then, if the feature is not enabled, the benchmark won't be compiled.

Reply:
It works! I spent waaaaay too much time trying to get that to work, tyvm! :)